crates/solidity/outputs/cargo/crate/src/backend/passes/p1_flatten_contracts.rs

```diff
@@ -0,0 +1,69 @@
+use std::collections::HashMap;
+use std::rc::Rc;
+
+use super::p0_build_ast::Output as Input;
+use crate::backend::l2_flat_contracts::transformer::Transformer;
+use crate::backend::l2_flat_contracts::{
+    input, ContractDefinition, ContractDefinitionStruct, SourceUnit,
+};
+use crate::compilation::CompilationUnit;
+
+pub struct Output {
+    pub compilation_unit: CompilationUnit,
+    pub files: HashMap<String, SourceUnit>,
+}
+
+pub fn run(input: Input) -> Output {
+    let mut pass = Pass {};
+    let files = input
+        .files
+        .iter()
+        .map(|(file_id, source_unit)| (file_id.clone(), pass.transform_source_unit(source_unit)))
+        .collect();
+    let compilation_unit = input.compilation_unit;
+    Output {
+        compilation_unit,
+        files,
+    }
+}
+
+struct Pass {}
+
+impl Transformer for Pass {
+    fn transform_contract_definition(
+        &mut self,
+        source: &input::ContractDefinition,
+    ) -> ContractDefinition {
+        let node_id = source.node_id;
+        let abstract_keyword = source.abstract_keyword.as_ref().map(Rc::clone);
+        let name = Rc::clone(&source.name);
+        let members = self.transform_contract_members(&source.members);
+        let inheritance_types = source
```

Comment:
I'm failing to see how this is flattening the structure, but perhaps with a better test (see below) I can be convinced that this is doing it.